src/tests.rs

```diff
@@ -1,279 +1,163 @@
+        use crate::cli::cli_add_task;
+        use crate::{db::open_db, types::Task};
 
+        let task = Task::from(String::from("update excel sheet #test @today .5"));
         let db = open_db();
+
         db.create_table_if_missing();
+        cli_add_task(&db, task.clone());
 
-        cli_add_task(&db, task);
-        let tasks = cli_get_tasks(&db, "#work".to_string());
-        let task = tasks.get(0).unwrap();
+        let tasks = cli_get_tasks(&db, "#test".to_string(), false);
+        let task1 = tasks.get(0).unwrap();
 
-        assert_eq!(task.title, "update excel sheet");
-        assert_eq!(task.tag, "#work");
-        assert_eq!(task.priority, 3);
+        assert_eq!(task.title, task1.title);
+        assert_eq!(task.tag, task1.tag);
+        assert_eq!(task.priority, task1.priority);
+        assert_eq!(task.due, task1.due);
+        assert_eq!(task.finished, task1.finished);
 
         db.con.close().expect("Closing Database failed.");
     }
 
     #[test]
     fn get_task_by_prio() {
-        let r = parse_args(vec![
-            "opus".to_string(),
-            "add".to_string(),
-            "update excel sheet #work @today ,,,".to_string(),
-        ]);
-        let task = r.input.task.unwrap();
+        use crate::cli::cli_add_task;
+        use crate::{db::open_db, types::Task};
 
+        let task = Task::from(String::from("update excel sheet #work @today .18"));
         let db = open_db();
+
         db.create_table_if_missing();
+        cli_add_task(&db, task.clone());
 
-        cli_add_task(&db, task);
-        let tasks = cli_get_tasks(&db, ",,,".to_string());
-        let task = tasks.get(0).unwrap();
+        let tasks = cli_get_tasks(&db, ".18".to_string(), false);
+        let task1 = tasks.get(0).unwrap();
 
-        assert_eq!(task.title, "update excel sheet");
-        assert_eq!(task.tag, "#work");
-        assert_eq!(task.priority, 3);
+        assert_eq!(task.title, task1.title);
+        assert_eq!(task.tag, task1.tag);
+        assert_eq!(task.priority, task1.priority);
+        assert_eq!(task.due, task1.due);
+        assert_eq!(task.finished, task1.finished);
 
         db.con.close().expect("Closing Database failed.");
     }
 
     #[test]
     fn finish_task() {
-        let r = parse_args(vec![
-            "opus".to_string(),
-            "add".to_string(),
-            "update excel sheet #work @today ,,,".to_string(),
-        ]);
-        let task = r.input.task.unwrap();
+        use crate::cli::cli_add_task;
+        use crate::{db::open_db, types::Task};
+
+        let task = Task::from(String::from("update excel sheet #work @today .5"));
         let db = open_db();
 
         db.create_table_if_missing();
+        cli_add_task(&db, task.clone());
 
-        cli_add_task(&db, task);
-        let id = db.con.last_insert_rowid().to_string();
+        let id = db.con.last_insert_rowid();
 
-        cli_fin_task(&db, id.clone());
+        cli_fin_task(&db, id.clone().to_string());
```

Comment:
The clone isn't necessary here when using `to_string()` because it doesn't consume the number.
```suggestion
        cli_fin_task(&db, id.to_string());
```